### ISSUE TITLE
Triton client InferenceServerException: Failed to process request due to TypeError

### ISSUE THREAD

**Issue by benhgm** [triton-inference-server/server#5867]
**Description**
I am getting a TypeError when calling the Triton Inference Server using the httpclient.InferenceServerClient().infer() method. See screenshot below
![image](https://github.com/triton-inference-server/server/assets/80461537/25a822a3-5f37-4066-9525-0550e75c9a74)

**Background*
I am attempting to host a python-backed model on triton

**Triton Information**
21.08 (nvcr.io/nvidia/tritonserver:21.08-py3)

Are you using the Triton container or did you build it yourself?

**To Reproduce**
Model Definition is as follows
```
import os
import glob
import torch
import torch.nn as nn
from os import path
from PIL import Image
from torchvision import transforms as T

import tritonclient.http as httpclient
import triton_python_backend_utils as pb_utils

class Preprocess(nn.Module):
    """
    Define a preprocessing module to perform preprocessing of images before inference
    """
    def __init__(self):
        super().__init__()
        self.transforms = T.Compose([
                T.Resize(size=(288, 144)),
                T.ToTensor(),
                T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
                ])
    
    def forward(self, x):
        x = self.transforms(x)
        x = x.unsqueeze(0)
        return x
    
class TritonPythonModel:
    
    def initialize(self, args):
        """
        `initialize` is called only once when the model is being loaded.
        Implementing `initialize` function is optional. This function allows
        the model to intialize any state associated with this model.

        Parameters
        ----------
        args : dict
          Both keys and values are strings. The dictionary keys and values are:
          * model_config: A JSON string containing the model configuration
          * model_instance_kind: A string containing model instance kind
          * model_instance_device_id: A string containing model instance device ID
          * model_repository: Model repository path
          * model_version: Model version
          * model_name: Model name
        """
        
        print("Initialized...")
        
    def execute(self, requests):
        """
        execute` MUST be implemented in every Python model. `execute`
        function receives a list of pb_utils.InferenceRequest as the only
        argument. This function is called when an inference request is made
        for this model. Depending on the batching configuration (e.g. Dynamic
        Batching) used, `requests` may contain multiple requests. Every
        Python model, must create one pb_utils.InferenceResponse for every
        pb_utils.InferenceRequest in `requests`. If there is an error, you can
        set the error argument when creating a pb_utils.InferenceResponse

        Parameters
        ----------
        requests : list
          A list of pb_utils.InferenceRequest

        Returns
        -------
        list
          A list of pb_utils.InferenceResponse. The length of this list must
          be the same as `requests`
        """
        self.preprocess_model = Preprocess()
        
        responses = []
        
        for request in requests:
            in_0 = pb_utils.get_input_tensor_by_name(request, "INPUT_0")
            in_0 = in_0.as_numpy()
            
            out_0 = self.preprocess_model(in_0)
            
            out_tensor_0 = pb_utils.Tensor("OUTPUT_0",
                                           out_0.as_type(np.float32))
            
            inference_response = pb_utils.InferenceResponse(
                output_tensors=[out_tensor_0]
            )
            responses.append(inference_response)
        
        return responses 

    def finalize(self):
        """
        `finalize` is called only once when the model is being unloaded.
        Implementing `finalize` function is OPTIONAL. This function allows
        the model to perform any necessary clean ups before exit.
        """
        print("Cleaning up...")
```

Client file is as follows
```
from tritonclient.utils import *
import tritonclient.http as httpclient
import sys
import numpy as np

model_name = 'preprocess'
shape = [4]

with httpclient.InferenceServerClient("localhost:8000") as client:
    input0_data = np.random.rand(*shape).astype(np.uint8)
    inputs = [httpclient.InferInput("INPUT_0", input0_data.shape, np_to_triton_dtype(input0_data.dtype))]
    inputs[0].set_data_from_numpy(input0_data)
    
    outputs = [httpclient.InferRequestedOutput("OUTPUT_0")]
    
    response = client.infer(
        model_name=model_name,
        inputs=inputs,
        outputs=None
    )
    
    result = response.get_response()
    output0_data = response.as_numpy("OUTPUT_0")
    
    print("PASS")
    sys.exit(0)
    
```

**Expected behavior**
Able to host the model without errors

**Comment by Tabrizian** [comment]
Can you also share the line number shared in the logs? Not sure whether 21.08 logged the line number.

**Comment by benhgm** [comment]
> Can you also share the line number printed in the logs? I'm not sure whether 21.08 logged the line number but in case it logs it that would be helpful for finding the root cause.

Hi @Tabrizian I found out the issue was in my model definition. The data structure that I passed into my torch.nn.Module class is not correct. To answer your question, 21.08 does not print line number where it fails in the model. It only shows the error point in the client.py file.

How do I activate detailed logs on tritonserver so that I can debug such issues better?
